src/Microsoft.AspNetCore.Localization.Routing/RouteDataRequestCultureProvider.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+
+namespace Microsoft.AspNetCore.Localization.Routing
+{
+    /// <summary>
+    /// Determines the culture information for a request via values in the route data.
+    /// </summary>
+    public class RouteDataRequestCultureProvider : RequestCultureProvider
+    {
+        /// <summary>
+        /// The key that contains the culture name.
+        /// Defaults to "culture".
+        /// </summary>
+        public string RouteDataStringKey { get; set; } = "culture";
+
+        /// <summary>
+        /// The key that contains the UI culture name. If not specified or no value is found,
+        /// <see cref="RouteDataStringKey"/> will be used.
+        /// Defaults to "ui-culture".
+        /// </summary>
+        public string UIRouteDataStringKey { get; set; } = "ui-culture";
```

Comment:
Experience here is in line with existing culture providers. So I do not want to do anything different  with this new provider.